reactor-core/src/test/java/reactor/core/publisher/FluxPeekFuseableTest.java

```diff
@@ -493,7 +493,7 @@ public void syncFusionAvailable() {
 		    .subscribe(ts);
 
 		Subscription s = ts.upstream();
-		assertThat(s).as("Non-fuseable upstream: " + s).isInstanceOf(QueueSubscription.class);
+		assertThat(s).as("Non-fuseable upstream").isInstanceOf(QueueSubscription.class);
```

Comment:
wondering about the removal of `s` here. can we make sure the message displays the toString of  `s` anyway?

Reply:
It does, but there is actually another problem: assertJ tries to be smart when printing `s` and tries to call `iterator()` on it. And so we get
```
Although QueueSubscription extends Queue it is purely internal and only guarantees support for poll/clear/size/isEmpty. Instances shouldn't be used/exposed as Queue outside of Reactor operators.
```
which is a design problem in the first place IMHO (Liskov substitution principle violation)

Reply:
`withFailMessage` should fix that I think